batch2/batch/driver.py

```diff
@@ -0,0 +1,470 @@
+import os
+import json
+import asyncio
+import aiohttp
+import logging
+import google.oauth2.service_account
+import sortedcontainers
+import traceback
+from aiohttp import web
+
+from hailtop.config import get_deploy_config
+from hailtop.utils import AsyncWorkerPool
+
+from .google_compute import GServices
+from .instance_pool import InstancePool
+from .utils import parse_cpu
+from .globals import get_db, tasks
+
+log = logging.getLogger('driver')
+
+db = get_db()
+
+
+class DriverException(Exception):
+    def __init__(self, status, message):
+        self.status = status
+        self.message = message
+
+    def __str__(self):
+        return f'{(self.status, self.message)}'
+
+
+class Pod:
+    @staticmethod
+    def from_record(driver, record):
+        spec = json.loads(record['spec'])
+        status = json.loads(record['status']) if record['status'] else None
+
+        inst = driver.inst_pool.token_inst.get(record['instance'])
```

Comment:
As a general comment, I think we should prefer the full name `instance` since it's not really that long and is more clear.